shell/platform/darwin/macos/framework/Source/FlutterSurface.mm

```diff
@@ -10,11 +10,22 @@ @interface FlutterSurface () {
   CGSize _size;
   IOSurfaceRef _ioSurface;
   id<MTLTexture> _texture;
+  int _age;
+  // Used for testing.
+  BOOL _isInUseOverride;
 }
 @end
 
 @implementation FlutterSurface
 
+- (void)setAge:(int)age {
+  self->_age = age;
+}
+
+- (int)age {
+  return _age;
+}
```

Comment:
Since the surfaces themselves don't need to know about this info, just the surface manager, instead of adding the public interface (granted, on an internal class), consider storing this info in the surface manager itself (e.g. in a map).